### DISCUSSION TITLE
Request log

### DISCUSSION THREAD

**Discussion by cdegroc** [JanusGraph/janusgraph#2300] [Q&A]
👋  I would like to log/dump all requests JanusGraph server receives to use them for benchmarking.
What would be the best way to achieve this? (e.g. a logger config in netty/tinkerpop/janusgraph, transaction log).
Thanks in advance.

**Comment by cdegroc** [comment]
This post on gremlin-users ML [suggests using the auditLogger](https://groups.google.com/g/gremlin-users/c/W6Jtq0s_jT4/m/SsE_OK8tBQAJ).
However the audit logger (for `TraversalOpProcessor`) is not available in branch 3.4-dev and only master.
One might have to wait for TinkerPop 3.5.0 to log requests without a creating a custom image of TinkerPop/JanusGraph.

**Comment by BrunoBerisso** [comment]
I'm just brainstorming here so take this with a grain of salt (or two).

One option could be to extend one of the default `Channelizer` implementations and add a custom handler to log the requests, something similar to this: https://github.com/apache/tinkerpop/blob/master/gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/channel/WebSocketChannelizer.java#L85 The downside is that you will receive the raw request so you will need an extra parsing to get useful information.

If you are in control of the requests another option could be the [Trigger Log](https://docs.janusgraph.org/advanced-topics/janusgraph-bus/#trigger-log).

**Reply by cdegroc** [reply]
Thanks Bruno.

I came to the same conclusion regarding overriding the Channelizer but didn't want to deserialize twice.
For the Trigger Log approach, I'd have to be able to create a transaction (with `g.buildTransaction()`). I might have missed it, but that didn't seem doable from gremlin-python.

I ended up monkey patching `TraversalOpProcessor` to log requests from there (and created [TINKERPOP-2523](https://issues.apache.org/jira/projects/TINKERPOP/issues/TINKERPOP-2523) to hopefully do it more cleanly in the future).
